### PR TITLE
added propTypes/fixed badge deleting

### PR DESCRIPTION
**Pull request by VArsenyan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React, { useEffect } from "react";
 import { Switch } from "react-router";
 import { connect } from "react-redux";
+import PropTypes from "prop-types";
+
```
Comment (Levon7707) [inline src/App.js]: Remove line break

File: src/Components/AddSection/AddBadge/AddBadge.jsx [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import React, { useState } from "react";
 import { connect } from "react-redux";
+import PropTypes from "prop-types";
```
Comment (Levon7707) [inline src/Components/AddSection/AddBadge/AddBadge.jsx]: Don't have line breaks between imports

File: src/Components/AddSection/AddIssue/AddIssue.jsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import React, { useState } from "react";
 import { NavLink } from "react-router-dom";
 import { connect } from "react-redux";
+import PropTypes from "prop-types";
 import {
   setIssues,
   setSelectedBadges,
   delSelectedBadges,
   clearSelectedBadges,
 } from "../../../Actions/actions";
```
Comment (Levon7707) [inline src/Components/AddSection/AddIssue/AddIssue.jsx]: line break

File: src/Components/AddSection/AddIssue/AddIssue.jsx [hunk 4]
```diff
@@ -64,38 +63,33 @@ const AddIssue = ({
     }
   };
 
-  // todo fixel bug@ vor avelacnum a mi qani label u
   const addBadges = (b) => {
```
Comment (Levon7707) [inline src/Components/AddSection/AddIssue/AddIssue.jsx]: b is bad variable name, change to badge for example

File: src/Components/AddSection/AddIssue/AddIssue.jsx [hunk 5]
```diff
@@ -64,38 +63,33 @@ const AddIssue = ({
...
-          key={b.id}
-          className="badge"
-          onClick={() => delBadges(b)}
-          style={{ backgroundColor: b.color }}
-        >
-          {b.text}
-          <i className="fas fa-minus" />
-        </p>
+    const renderBadges = () => {
+      return selectedBadges.map((b) => (
```
Comment (Levon7707) [inline src/Components/AddSection/AddIssue/AddIssue.jsx]: change b to badge

File: src/Components/AddSection/AddIssue/AddIssue.jsx [hunk 6]
```diff
@@ -157,8 +151,18 @@ const AddIssue = ({
...
+  badges: PropTypes.array.isRequired,
+  length: PropTypes.number.isRequired,
+  selectedBadges: PropTypes.array.isRequired,
+  setIssues: PropTypes.func.isRequired,
+  setSelectedBadges: PropTypes.func.isRequired,
+  delSelectedBadges: PropTypes.func.isRequired,
+  clearSelectedBadges: PropTypes.func.isRequired,
+};
+
 const mapStateToProps = (state) => ({
```
Comment (Levon7707) [inline src/Components/AddSection/AddIssue/AddIssue.jsx]: remove Reducer from all reducer variables.
state.badgesReducer => state.badges.
No need to have the reducer in the name, as we know that in state it is reducer

File: src/Components/IssuesList/AllIssuesList.jsx [hunk 7]
```diff
@@ -1,5 +1,7 @@
 import React from "react";
 import { connect } from "react-redux";
+import PropTypes from "prop-types";
```
Comment (Levon7707) [inline src/Components/IssuesList/AllIssuesList.jsx]: line  break

File: src/Components/IssuesList/FindedIssuesList.jsx [hunk 8]
```diff
@@ -1,5 +1,7 @@
 import React from "react";
 import { connect } from 'react-redux';
+import PropTypes from "prop-types";
```
Comment (Levon7707) [inline src/Components/IssuesList/FindedIssuesList.jsx]: line break

File: src/Components/IssuesList/IssuesListHeader/IssuesListHeader.jsx [hunk 9]
```diff
@@ -1,9 +1,11 @@
 import React, { useState } from "react";
 import { NavLink } from "react-router-dom";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
```
Comment (Levon7707) [inline src/Components/IssuesList/IssuesListHeader/IssuesListHeader.jsx]: line break

File: src/Components/IssuesList/SortedIssuesList.jsx [hunk 10]
```diff
@@ -1,5 +1,7 @@
 import React from "react";
 import { connect } from "react-redux";
+import PropTypes from "prop-types";
```
Comment (Levon7707) [inline src/Components/IssuesList/SortedIssuesList.jsx]: line break

File: src/Components/NewIssueHeader/NewIssueHeader.jsx [hunk 11]
```diff
@@ -58,6 +59,11 @@ const NewIssueHeader = ({ findByName }) => {
   );
 };
 
+NewIssueHeader.propTypes = {
+  findByName: PropTypes.func.isRequired,
+};
+
```
Comment (Levon7707) [inline src/Components/NewIssueHeader/NewIssueHeader.jsx]: remove line break

File: src/Components/SelectedIssue/SelectedIssue.jsx [hunk 12]
```diff
@@ -1,11 +1,14 @@
 import React from "react";
 import { connect } from "react-redux";
+import PropTypes from "prop-types";
+
 import AddComment from "./AddComment/AddComment";
 import Preloader from '../Preloader';
 
 import "./SelectedIssue.scss";
 
 const SelectedIssue = ({ selectedIssue }) => {
```
Comment (Levon7707) [inline src/Components/SelectedIssue/SelectedIssue.jsx:10]: remove console

File: src/Redux/badges-reducer.js [hunk 13]
```diff
@@ -6,26 +6,27 @@ import {
...
-      });
-      return state;
+      const newState = state.selectedBadges.filter( val => val.id !== action.delId.id );
+      return {
+        ...state,
+        selectedBadges: newState,
+      };
     case CLEAR_SELECTED_BADGE:
-      state.selected_badges = [];
+      state.selectedBadges = [];
```
Comment (Levon7707) [inline src/Redux/badges-reducer.js]: don't change state directly

File: src/App.js (hunk 1 above)
Comment (VArsenyan) [reply src/App.js]: fixed

File: src/Components/AddSection/AddBadge/AddBadge.jsx (hunk 2 above)
Comment (VArsenyan) [reply src/Components/AddSection/AddBadge/AddBadge.jsx]: fixed

File: src/Components/AddSection/AddIssue/AddIssue.jsx (hunk 3 above)
Comment (VArsenyan) [reply src/Components/AddSection/AddIssue/AddIssue.jsx]: fixed

File: src/Components/AddSection/AddIssue/AddIssue.jsx (hunk 4 above)
Comment (VArsenyan) [reply src/Components/AddSection/AddIssue/AddIssue.jsx]: fixed

File: src/App.js (hunk 1 above)
Comment (Levon7707) [reply src/App.js]: @VArsenyan not fixed

**Pull request merged**